interface/grpc/core/list_services.go

```diff
@@ -3,14 +3,53 @@ package core
 import (
 	"context"
 
+	"github.com/mesg-foundation/core/api"
 	"github.com/mesg-foundation/core/protobuf/coreapi"
+	service "github.com/mesg-foundation/core/service"
+	"github.com/mesg-foundation/core/x/xerrors"
 )
 
 // ListServices lists services.
 func (s *Server) ListServices(ctx context.Context, request *coreapi.ListServicesRequest) (*coreapi.ListServicesReply, error) {
-	services, err := s.api.ListServices()
+	var filters []api.ListServicesFilter
+	if request.FilterActive {
+		filters = append(filters, api.ListRunningServicesFilter())
+	}
+	services, err := s.api.ListServices(filters...)
 	if err != nil {
 		return nil, err
 	}
-	return &coreapi.ListServicesReply{Services: toProtoServices(services)}, nil
+
+	protoServices := toProtoServices(services)
+
+	var (
+		servicesLen = len(services)
+		errC        = make(chan error, servicesLen)
+	)
+
+	// fill services status info.
+	for _, s := range services {
```

Comment:
done. i made it that way to have a logical separation while updating _Service_ fields in case we do set additional fields like this in future. but i agree and i'd rather reducing the complexity of code.